src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_docker_utils.py

```diff
@@ -176,9 +176,12 @@ def _get_credentials(cmd,  # pylint: disable=too-many-statements
             registry_name, '-{}'.format(tenant_suffix) if tenant_suffix else '', login_server_suffix).lower()
 
     # Validate the login server is reachable
-    challenge = 'https://' + login_server + '/v2/'
+    url = 'https://' + login_server + '/v2/'
     try:
-        requests.get(challenge, verify=(not should_disable_connection_verify()))
+        challenge = requests.get(url, verify=(not should_disable_connection_verify()))
```

Comment:
@yuwaMSFT2 All runtime requests (including classic) will first run through this check to make sure it's reachable and accessible.